### PR TITLE
Tree mapper and Account History!!!

### PR DESCRIPTION
**Pull request by rishsingh07**
To Do - 

- [x] Certificates - Account History
- [x] dot markers for sample trees - Treemapper
- [x]  use env variable for CDN: https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1022#discussion_r627960285 or https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1022#discussion_r639569703
- [x] merge conflicts
- [ ] scroll issue with open account record (desktop)
- [ ] verify the backend response with notion doc here - https://www.notion.so/plantfortheplanet/Account-History-API-2ac912ccd19743928b303386ccdae2e7
- [ ] Problem showing profile page with none develop build: https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1022#issuecomment-850434765

To Test - 

- [ ] Register Trees
- [ ] Manage Projects
- [ ] Add Projects
- [ ] logout on all pages
- [ ] visit all pages not logged in

### CODE REVIEW

**Comment by sarveshpro** [comment]
- [ ] The text inside the header should not go outside 1200px content width. please refer accounts page.
- [ ] The settings button is missing.
- [ ] The background of the form and the page mismatch. please use $backgroundColor variable everywhere.
- [ ] get rid of the footer but leave some space(equal to the height of the footer) empty at the bottom of the page.
- [ ] use same header component for accounts page

image

**Review by sarveshpro — CHANGES_REQUESTED** [review]
Please make the above changes

File: src/features/common/Layout/Header/accountHeader.tsx
Comment (norbertschuler) [inline src/features/common/Layout/Header/accountHeader.tsx:71]: What is the difference between specifying the `redirectUri` directly in the `loginWithRedirect` function call or setting them in our localStorage like at https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/f358ad2e551ba81584c090d0ca88c105c59f0b09/pages/manage-projects/add-project/index.tsx#L53-L54?

Comment (norbertschuler) [reply src/features/common/Layout/Header/accountHeader.tsx:71]: I assume if we want to support a redirectUri for Auth0, we need to add it to all configurations at Auth0 (which could be a lot of work as we already have added quite a lot of applications). Currently `/account/history` is not configured resulting into an error, so we probably need to change this towards using the solution I linked above:

Bildschirmfoto 2021-04-28 um 09 03 48

File: pages/manage-projects/[id].tsx
Comment (norbertschuler) [inline pages/manage-projects/[id].tsx:11]: Please remove unused imports.

File: public/static/locales/en/me.json
Comment (norbertschuler) [inline public/static/locales/en/me.json:6]: We should try to avoid repeating too much duplicate content in translation resources (at least if the translation is within the same resource file). This is already existing as https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/f358ad2e551ba81584c090d0ca88c105c59f0b09/public/static/locales/en/me.json#L11

File: src/features/common/Layout/Header/Header.module.scss
Comment (norbertschuler) [inline src/features/common/Layout/Header/Header.module.scss:21]: Is is possible to also try to take the value from our environment variable in an .scss file as in an .tsx file? Or can we rewrite this to load the background image in the related component using this style like we did it for the user profile at https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/f358ad2e551ba81584c090d0ca88c105c59f0b09/src/features/user/UserProfile/screens/IndividualProfile.tsx#L61-L63

File: src/features/common/Layout/Header/accountHeader.tsx
Comment (norbertschuler) [inline src/features/common/Layout/Header/accountHeader.tsx:71]: This code block looks like a duplicate the following code - can't we abstract it somehow? https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/f358ad2e551ba81584c090d0ca88c105c59f0b09/src/features/user/Account/screens/index.tsx#L73-L104  A similar code pattern also is used in https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/develop/src/utils/auth0/userInfo.ts.

**Review by norbertschuler — CHANGES_REQUESTED** [review]

File: src/features/common/Layout/Header/Header.module.scss
Comment (rishsingh07) [reply src/features/common/Layout/Header/Header.module.scss:21]: I guess individual profile uses different header. The header component just has back button and settings button which is going to be used in different pages which does not require other features which individual profile has.

File: src/features/common/Layout/Header/accountHeader.tsx
Comment (rishsingh07) [reply src/features/common/Layout/Header/accountHeader.tsx:71]: Yess I forgot to remove it from accounts

File: pages/manage-projects/[id].tsx
Comment (rishsingh07) [reply pages/manage-projects/[id].tsx]: done

File: public/static/locales/en/me.json
Comment (rishsingh07) [reply public/static/locales/en/me.json:6]: done

File: src/features/common/Layout/Header/accountHeader.tsx
Comment (rishsingh07) [reply src/features/common/Layout/Header/accountHeader.tsx:71]: @sarveshpro can we remove it from accounts?

File: public/static/locales/en/me.json
Comment (norbertschuler) [reply public/static/locales/en/me.json:6]: Please do not just remove `settingManageProject` as it is of course used at https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/b49b0a05d22130b0f9345244937383d84eeecc8c/src/features/user/UserProfile/components/SettingsModal.tsx#L98

Comment (rishsingh07) [reply public/static/locales/en/me.json:6]: my bad, fixed.

Comment (norbertschuler) [reply public/static/locales/en/me.json]: I did some more fixes and removed the now unused resource `manageProjects`.

**Comment by norbertschuler** [comment]
- [ ] Adding new projects does not yet have a header, e.g. with http://localhost:3000/manage-projects/add-project

File: src/features/common/Layout/Header/accountHeader.tsx
Comment (sarveshpro) [reply src/features/common/Layout/Header/accountHeader.tsx:71]: I'll suggest you create a userContext. that way you can update it's value from here and use it anywhere in the app.

Comment (norbertschuler) [reply src/features/common/Layout/Header/accountHeader.tsx:71]: I saw that someone added http://localhost:3000/account/history to the valid Callback URLs of our development Auth0 tenant `pftp`, but not for the production Auth0 tenant. Thus this will still fail on production. If we want to allow this URL as a callback URL, it has to be done everywhere!

Comment (norbertschuler) [reply src/features/common/Layout/Header/accountHeader.tsx]: There would be at least two production applications on Auth0 to adapt, but this would have to be maintained for every other application (tenant) allowing this URL in the future. I would prefer the code as following:

```
localStorage.setItem('redirectLink','/account/history');
loginWithRedirect({
  redirectUri: `${process.env.NEXTAUTH_URL}/login `,
  ui_locales: localStorage.getItem('language') || 'en',
});
```

File: src/features/common/Layout/Header/accountHeader.tsx
Comment (norbertschuler) [inline src/features/common/Layout/Header/accountHeader.tsx:49]: Going backwards the browser history can result into an error if going through the Auth0 login process for URLs like http://localhost:3000/manage-projects/add-project or http://localhost:3000/account/history as following:

Bildschirmfoto 2021-04-30 um 16 12 18

**Review by norbertschuler — CHANGES_REQUESTED** [review]

File: src/features/common/Layout/Header/accountHeader.tsx
Comment (norbertschuler) [reply src/features/common/Layout/Header/accountHeader.tsx]: I change it.

File: pages/_app.tsx
Comment (norbertschuler) [inline pages/_app.tsx:157]: @sarveshpro can we be sure the user handling in this abstracted HOC does not interfere with any other handling in our code e.g. https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/ec938bc0d35d5ac659fc1710bebb9f65446b2bb2/pages/t/%5Bid%5D.tsx#L90-L112 or https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/ec938bc0d35d5ac659fc1710bebb9f65446b2bb2/src/utils/auth0/userInfo.ts#L7-L46

Comment (norbertschuler) [reply pages/_app.tsx:157]: At least this change cost us one additional API call to fetch the user's profile. 

- [ ] We already have a duplication at `/login` following even by a third call when redirecting to the user's profile `/t/user` - now this is calling the profile API four(!) times.

Bildschirmfoto 2021-05-03 um 08 23 14

- [ ] The duplication now is calling the profile API call twice being logged in when opening the user's profile.

Bildschirmfoto 2021-05-03 um 08 24 06

**Comment by sarveshpro** [comment]
can someone test everything related to user flow on this branch?

- [ ] login
- [ ] logout
- [ ] signup
- [ ] public profile
- [ ] private profile
- [ ] All authenticated pages and API calls

File: src/features/common/Layout/Header/AccountHeader.module.scss
Comment (norbertschuler) [inline src/features/common/Layout/Header/AccountHeader.module.scss:16]: I still want to ask if it's possible to set this image within the JavaScript code to allow us to use the env variable CDN_URL for the URL in order to be able to support different environments.

**Review by norbertschuler — CHANGES_REQUESTED** [review]

File: src/features/common/Layout/Header/Header.module.scss
Comment (norbertschuler) [reply src/features/common/Layout/Header/Header.module.scss]: close in favor of https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1022#discussion_r627960285

File: src/features/common/Layout/Header/AccountHeader.module.scss
Comment (sarveshpro) [reply src/features/common/Layout/Header/AccountHeader.module.scss:16]: Yes, we could do it but the CDN for development doesn't have this image 😅

Comment (norbertschuler) [reply src/features/common/Layout/Header/AccountHeader.module.scss:16]: @sagararyal can we deploy above image to all CDN (develop, staging, production)?

File: src/features/common/Layout/Header/accountHeader.tsx
Comment (norbertschuler) [reply src/features/common/Layout/Header/accountHeader.tsx:51]: This should be solved in https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1057

File: src/features/common/Layout/Header/AccountHeader.module.scss
Comment (norbertschuler) [inline src/features/common/Layout/Header/AccountHeader.module.scss:44]: https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1022#discussion_r627960285 is still open, but the code got moved here.

**Review by norbertschuler — CHANGES_REQUESTED** [review]

**Comment by norbertschuler** [comment]
Both Vercel and Cloudflare Pages have build problem throwing this error:


11:03:43.390 | Error: [object Object] is not a PostCSS plugin
-- | --
11:03:43.390 | &gt; Build error occurred
11:03:43.391 | Error: &gt; Build failed because of webpack errors
11:03:43.391 | at /opt/buildhome/repo/node_modules/next/dist/build/index.js:17:924
11:03:43.391 | at runMicrotasks (&lt;anonymous&gt;)
11:03:43.391 | at processTicksAndRejections (internal/process/task_queues.js:97:5)
11:03:43.391 | at async Span.traceAsyncFn (/opt/buildhome/repo/node_modules/next/dist/telemetry/trace/trace.js:6:584)

**Comment by norbertschuler** [comment]
- [ ] I can not login any more currently - anyone else?

Bildschirmfoto 2021-05-28 um 15 51 05

**Comment by norbertschuler** [comment]
> * [x]  After login, the profile page does not load - pages as .e.g https://planet-webapp-9kp2wm9a6-planetapp.vercel.app/account/history can be loaded, so the login succeeded, but the profile e.g. https://planet-webapp-9kp2wm9a6-planetapp.vercel.app/t/norbert-schuler does not get loaded and instead showing an endless spinner:
> 
> Update: This also happening locally when using `npm run build & npm run start -p 3000` - but not if running `npm run dev`!

Still happening to me on Vercel:

111C5545-7248-4942-B108-5598800C1666

**Comment by sarveshpro** [comment]
> > * [x]   After login, the profile page does not load - pages as .e.g https://planet-webapp-9kp2wm9a6-planetapp.vercel.app/account/history can be loaded, so the login succeeded, but the profile e.g. https://planet-webapp-9kp2wm9a6-planetapp.vercel.app/t/norbert-schuler does not get loaded and instead showing an endless spinner:
> > 
> > Update: This also happening locally when using `npm run build & npm run start -p 3000` - but not if running `npm run dev`!
> 
> Still happening to me on Vercel:
> 
> 120145811-6b30da00-c1e4-11eb-8198-e2a682d63f3c

Apparently this is now working on local `npm run build` & `npm run start` but not on vercel deployment

**Comment by sarveshpro** [comment]
failed, moving to other tasks

**Comment by sarveshpro** [comment]
same issue happens on netlify - https://planet-sarvesh.netlify.app/

**Comment by harshvitra** [comment]
Reason why we don't seen anything in console because there is a compile time error

**Comment by sarveshpro** [comment]
This PR is a lost cause, closing it. Don't delete the branch immediately tho. wait until Treemapper and A/c history are merged to develop

**Pull request closed without merging**
